LongLookupTables/make_lookup_tables.py

```diff
@@ -0,0 +1,389 @@
+
+Help : `python make_lookup_tables.py -h`
+"""
+
+from __future__ import unicode_literals, division, absolute_import, print_function
+
+import sys
+import os
+import itertools
+import random
+import operator
+import warnings
+import argparse
+from functools import reduce
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+### MAIN ###
+def parse_arguments(args):
+    parser = argparse.ArgumentParser(description="Script to generate of the lookup tables problem.",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('-d', '--dir', default='.', help='Path to the directory where to save the generated data.')
+    parser.add_argument('-s', '--n-samples', type=int, default=1, help='Number of different samples to generate. If greater than 1, will save the files for a single sample in /dir/sample<i>/*.tsv.')
+
+    parser.add_argument('-v', '--validation-size', type=float, default=0.1, help='Percentage of training set to use as validation.')
+    parser.add_argument('-c', '--max-composition-train', type=int, default=4, help='Max length of compositions in training set.')
+    parser.add_argument('-t', '--n-unary-tables', type=int, default=8, help='Number of different lookup tables.')
+    parser.add_argument('-T', '--n-heldout-tables', type=int, default=2, help='Number of tables that would only be seen in unary.')
+    parser.add_argument('-C', '--n-heldout-compositions', type=int, default=50, help='Number of compositions to randomly remove.')
+    parser.add_argument('-I', '--n-heldout-inputs', type=int, default=2, help='Number of inputs to heldout from training.')
+    parser.add_argument('-l', '--n-longer', type=int, default=2, help='Number of additional tables to add to `longer` test data.')
+    parser.add_argument('--not-intermediate', action='store_true', help='Removes intermediate step in the target sequence.')
+    parser.add_argument('--not_shuffle', action='store_true', help='Disables shuffling of the outputed datasets')
+    parser.add_argument('--not_reverse', action='store_true', help='Disables reversing of the input sequence. I.e if given, then uses `t1∘t2 input` instead of `input t2 t1`.')
+    parser.add_argument('--not_stratify', action='store_true', help='Disables the balancing of the lookups in train and validation set.')
+    parser.add_argument('-e', '--eos', default='.', help='EOS token to append at the end of each input.')
+    parser.add_argument('-b', '--bound-longer', type=int, default=10**4, help='Bounds the maximum number of permuted compositions in `longer`')
+    parser.add_argument('-a', '--alphabet', metavar=('letter1', 'letter2'), nargs='*', default=['0', '1'], help='Possible characters given as input.')
+    parser.add_argument('-r', '--n-repeats', type=int, default=3, help='Number of characters in `alphabet` used in each input and output.')
```

Comment:
is this referring to what in the original formulation is referred to with 2bit, 3bit.., perhaps use a name which is closer to that?

Reply:
exactly alphabet is for now `['0','1']` so bits. And 2, 3 is `n_repeats`. I'll see if I find better names, if not I'll add a better helper.

Reply:
ok, I think it is sort of clear. Or maybe n_bits would be good? This is what it is called in the original paper I believe.